server/controllers/stock/depot.js

```diff
@@ -19,6 +19,56 @@ exports.getAvailableLotsByInventoryId = getAvailableLotsByInventoryId;
 exports.getExpiredLots = getExpiredLots;
 exports.createDistributions = createDistributions;
 exports.getStockExpirations = getStockExpirations;
+exports.createDepots = createDepots;
+exports.editDepots = editDepots;
+
+/**
+* POST /depots
+* Create a new depot in the database
+*
+* @function createDepots
+*/
+function createDepots(req, res, next) {
+  'use strict';
+
+  var query = 'INSERT INTO `depot` SET ?';
+  db.exec(query, [req.body])
+  .then(function () {
+    res.status(201).json({ id : req.body.uuid });
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* PUT /depots
+* Edit an existing depot in the database
+*
+* @function editDepots
+*/
+function editDepots(req, res, next) {
+  'use strict';
+
+  /** prevent update uuid */
+  if (req.body.uuid) { delete req.body.uuid; }
+
+  var query = 'UPDATE `depot` SET ? WHERE uuid = ?';
+  db.exec(query, [req.body, req.params.uuid])
+  .then(selectDepot)
+  .then(function (rows) {
+    if (!rows.length) { return next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+
+  function selectDepot(rows) {
```

Comment:
Why don't you make this function return a JSON object instead of an array?  It could also throw a NOT FOUND error inside the promise, and that would clean up your GET and UPDATE routes a little bit.